corehq/apps/domain/models.py

```diff
@@ -548,7 +549,13 @@ def _get_by_name(stale=False):
             if not isinstance(result, Domain):
                 # A stale view may return a result with no doc if the doc has just been deleted.
                 # In this case couchdbkit just returns the raw view result as a dict
-                return None
+                try:
+                    result = Domain.get(name)
```

Comment:
The code you linked to is just setting the domain name to the ID.

On 7 May 2015 at 12:27, Ethan Soergel <notifications@github.com> wrote:

> In corehq/apps/domain/models.py
> <https://github.com/dimagi/commcare-hq/pull/6517#discussion_r29840375>:
>
> > @@ -548,7 +549,13 @@ def _get_by_name(stale=False):
> >              if not isinstance(result, Domain):
> >                  # A stale view may return a result with no doc if the doc has just been deleted.
> >                  # In this case couchdbkit just returns the raw view result as a dict
> > -                return None
> > +                try:
> > +                    result = Domain.get(name)
>
> I think that's the case with domain snapshots used on the exchange. It's
> bonkers, I know.
>
> https://github.com/dimagi/commcare-hq/blob/6e54af331f1304ece69d08c8d6d8564d0ba90705/corehq/apps/domain/models.py#L657-L661
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dimagi/commcare-hq/pull/6517/files#r29840375>.
>



-- 
Simon Kelly
Senior Engineer | Dimagi South Africa